sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
+                b = base.__class__(Identity(base.shape[0]))
+                if b == Identity(base.shape[0]).as_explicit():
+                    return b
+                else:
+                    return Identity(base.shape[0])
+            except TypeError:
+                return Identity(base.shape[0])
+        elif isinstance(base, ZeroMatrix) and exp < 0:
```

Comment:
You should check `exp.is_negative`. `exp < 0` will raise TypeError if `exp` is symbolic.